### DISCUSSION TITLE
Topic partition is not isolated base on  strimzi  kafka operator

### DISCUSSION THREAD

**Discussion by hmProgramer** [strimzi/strimzi-kafka-operator#6316] [Q&A] [answered]
My kafka cluster base on the strimzi operator, and the kafka version is 2.7.1, protocol is SSL,the external expose is ingress 
now i found a problem 
 when I created a topic name  of 'hm-topic-02'  partition 6 and replicas 3  in one cluster(has 3 broker)

Then I create a topic with the same name on another cluster(has 7broker), and the partition is 9 and the replicas is 3,

Then i check the first cluster and it becomes that the partition is 9 and the replica is 3,  the reproduce detail as below:

bin/kafka-topics.sh --bootstrap-server broker-4239-0.strimzi.kafka.mid:443,broker-4239-1.strimzi.kafka.mid:443,broker-4239-2.strimzi.kafka.mid:443 --create --topic hm-topic-02 --partitions 6 --replication-factor 3 --config max.message.bytes=64000 --config flush.messages=1 --command-config /usr/local/kafka_2.12-2.2.0/jks/4239.conf
[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4239-0.strimzi.kafka.mid:443,broker-4239-1.strimzi.kafka.mid:443,broker-4239-2.strimzi.kafka.mid:443 --topic hm-topic --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4239.conf

[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4239-0.strimzi.kafka.mid:443,broker-4239-1.strimzi.kafka.mid:443,broker-4239-2.strimzi.kafka.mid:443 --topic hm-topic-02 --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4239.conf
Topic:hm-topic-02       PartitionCount:6        ReplicationFactor:3     Configs:flush.messages=1,message.format.version=2.7-IV2,max.message.bytes=64000
        Topic: hm-topic-02      Partition: 0    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic-02      Partition: 1    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic-02      Partition: 2    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic-02      Partition: 3    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic-02      Partition: 4    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic-02      Partition: 5    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1

[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4248-0.strimzi.kafka.mid:443,broker-4248-1.strimzi.kafka.mid:443,broker-4248-2.strimzi.kafka.mid:443 --create --topic hm-topic-02 --partitions 9 --replication-factor 3 --config max.message.bytes=64000 --config flush.messages=1 --command-config /usr/local/kafka_2.12-2.2.0/jks/4248.conf
[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4239-0.strimzi.kafka.mid:443,broker-4239-1.strimzi.kafka.mid:443,broker-4239-2.strimzi.kafka.mid:443 --topic hm-topic-02 --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4239.conf
Topic:hm-topic-02       PartitionCount:9        ReplicationFactor:3     Configs:flush.messages=1,message.format.version=2.7-IV2,max.message.bytes=64000
        Topic: hm-topic-02      Partition: 0    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic-02      Partition: 1    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic-02      Partition: 2    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic-02      Partition: 3    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic-02      Partition: 4    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic-02      Partition: 5    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic-02      Partition: 6    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic-02      Partition: 7    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic-02      Partition: 8    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2

[root@Centos64 kafka_2.12-2.2.0]# bin/kafka-topics.sh --bootstrap-server broker-4248-0.strimzi.kafka.mid:443,broker-4248-1.strimzi.kafka.mid:443,broker-4248-2.strimzi.kafka.mid:443 --topic hm-topic-02 --describe  --command-config /usr/local/kafka_2.12-2.2.0/jks/4248.conf
Topic:hm-topic-02       PartitionCount:9        ReplicationFactor:3     Configs:flush.messages=1,message.format.version=2.7-IV2,max.message.bytes=64000
        Topic: hm-topic-02      Partition: 0    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic-02      Partition: 1    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic-02      Partition: 2    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
        Topic: hm-topic-02      Partition: 3    Leader: 2       Replicas: 2,0,1 Isr: 2,0,1
        Topic: hm-topic-02      Partition: 4    Leader: 1       Replicas: 1,2,0 Isr: 1,2,0
        Topic: hm-topic-02      Partition: 5    Leader: 0       Replicas: 0,1,2 Isr: 0,1,2
        Topic: hm-topic-02      Partition: 6    Leader: 2       Replicas: 2,1,0 Isr: 2,1,0
        Topic: hm-topic-02      Partition: 7    Leader: 1       Replicas: 1,0,2 Isr: 1,0,2
        Topic: hm-topic-02      Partition: 8    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1

Does anyone know what's going on？

**Comment by scholzj** [comment, marked as answer]
I think you will need to provide more information about how you run your Kafka clusters.

One thing to keep in mind is that each cluster should preferably have its own namespace. And if you have multiple clusters in a single namespace, you should make sure they do not have all Topic Operator enabled (only one Topic Operator can run per namespace to work properly).

**Reply by hmProgramer** [reply]
thanks very much, When I disable topic operator, this problem is solved
